crates/core/tedge_agent/src/http_rest.rs

```diff
@@ -0,0 +1,302 @@
+use futures::StreamExt;
+use hyper::{server::conn::AddrIncoming, Request};
+use hyper::{Body, Response, Server};
+use routerify::{Router, RouterService};
+use std::{io::Write, path::Path};
+use std::{net::SocketAddr, path::PathBuf};
+
+use tedge_utils::paths::create_directories;
+use tokio::io::AsyncReadExt;
+
+use crate::error::AgentError;
+const FILE_TRANSFER_HTTP_ENDPOINT: &str = "/tedge/*";
```

Comment:
Please be more organized. Such constants are defined here and there (as `FILE_TRANSFER_ROOT_PATH` in `agent.rs`), then not use (as in `create_directory_with_user_group("/var/tedge/file-transfer", tedge, tedge)`. Something for the bind address `127.0.0.1:3000` and the `tedge` url prefix. Furthermore the very next task will be to have these be configured somehow.